easybuild/easyconfigs/m/McXtrace/McXtrace-3.5.24-foss-2023a.eb

```diff
@@ -0,0 +1,44 @@
+# Author: Erik B Knudsen
+# United Neux
+
+easyblock = 'CMakeMake'
+
+name = 'McXtrace'
+version = '3.5.24'
+
+homepage = 'https://mcxtrace.org'
+description = "McXtrace is a Monte Carlo ray tracing X-ray instrumentaton and experiment simulation packages."
+
+toolchain = {'name': 'foss', 'version': '2023a'}
+
+github_account = 'mccode-dev'
+source_urls = ["https://github.com/mccode-dev/McCode/archive/refs/tags/"]
+sources = ['v%(version)s.tar.gz']
+checksums = ['b1958a24b6b7e06e65ff3fc302da6d306412fdaee15ac77fadf841e1521b773a']
+
+builddependencies = [
+    ('CMake', '3.26.3'),
+    ('Bison', '3.8.2'),
+    ('flex', '2.6.4'),
+]
+
+dependencies = [
+    ('Python', '3.11.3'),
+    ('SciPy-bundle', '2023.07'),
+    ('PyYAML', '6.0')
+]
+
+configopts = '-DBUILD_MCXTRACE=ON -DENSURE_NCRYSTAL=off -DMCVERSION=%(version)s -DBUILD_TOOLS=off -DBUILD_MCXRUN=on'
```

Comment:
For software like this, i would probably

1. hope their cmake scripts allow for an override to point to locally checkout sources
2. add all the sub-sources, possible with custom extract command or preconfigopts to put them in the right paths

failing 1. i would make a patch to comment out the downloading steps, and just place the sources where they are needed. You can use git_config source downloads with EB if you need particular commits. We strongly prefer to never just "fetch latest master" as it throws out all hope of reprodicibility.